app/controllers/favorites_controller.rb

```diff
@@ -2,7 +2,7 @@ class FavoritesController < ApplicationController
   before_action :require_user
 
   def index
-    @user = User.find(params[:user_id])
+    @user = User.find_by(username: params[:username])
```

Comment:
Really cool and now we're not exposing IDs.  Wonder why that was never brought up in M2 or before this time in M3.